opendg/data/data.py

```diff
@@ -0,0 +1,328 @@
+                    edge_dict[t] = [(u,v)]
+                else:
+                    edge_dict[t].append((u,v))
+        return edge_dict
+
+
+
+    def __getattr__(self, key: str) -> Any:
+        return getattr(self._store, key)
+    
+    def __setattr__(self, key: str, value: Any):
+        setattr(self._store, key, value)
+
+    def __delattr__(self, key: str):
+        delattr(self._store, key)
+
+    def __getitem__(self, key: str) -> Any:
+        return self._store[key]
+
+    def __setitem__(self, key: str, value: Any):
+        self._store[key] = value
+
+    def __delitem__(self, key: str):
+        if key in self._store:
+            del self._store[key]
+
+    def __copy__(self):
+        return CTDG(self._store.copy())
+
+    def __deepcopy__(self, memo):
+        return CTDG(self._store.copy())
+
+    def __repr__(self) -> str:
+        return f"CTDG({self._store.keys()})"
+    
+    ###########################################################################
+
+    def aggregate_graph(self, start_time:int=None, end_time:int=None) -> torch.Tensor:
+        r"""Aggregates the graph between start_time and end_time, inclusive.
+        Args:
+            start_time (int, optional): The start time of the aggregation. (default: :obj:`None`)
+            end_time (int, optional): The end time of the aggregation. (default: :obj:`None`)
+        Returns:
+            :class:`torch.Tensor`: The aggregated edge index tensor."""
+        if (start_time is None):
+            start_time = self.min_time
+        if (end_time is None):
+            end_time = self.max_time
+        assert start_time <= end_time, "start_time should be less or equal to end_time"
+        assert isinstance(start_time, int) and isinstance(end_time, int), "start_time and end_time should be integers"
+        out_index = self._store['events'][start_time].edges
+        for t in self._store['events'].keys():
+            if (t > start_time and t <= end_time):
+                out_index = torch.cat((out_index, self._store['events'][t].edges), dim=1)
+        return out_index
+
+    def to_events(self) -> Any:
+        r"""Converts a continuous time dynamic graph to a list of edge events.
+        Returns: a tuple of timestamps and edge index tensors."""
+        # Implement this method
+        edge_index = self._store['events'][self.min_time].edges
+        timestamps = [self.min_time]*(edge_index.shape[1])
+        for t in self._store['events'].keys():
+            if (t> self.min_time):
+                edge_index = torch.cat((edge_index, self._store['events'][t].edges), dim=1)
+                timestamps.extend([t]*(self._store['events'][t].edges.shape[1]))
+        timestamps = torch.tensor(timestamps)
+        return timestamps, edge_index
+    
+    def to_snapshots(self) -> Any:
+        """Converts a continuous time dynamic graph to a list of snapshots."""
+        # Implement this method
+        pass
+
+    def load_csv(self, file_path: str) -> Any:

```

Comment:
I think it's not very clear why we would want someone to be able to `load_csv` on an already initiated `TG` object. This would basically just wipe out the previous data, shouldn't this just be a new object?